src/xslt-polyfill-src.js

```diff
@@ -323,6 +330,25 @@
     }
   }
 
+  function replaceCurrentXMLDoc() {
+    const doc = new XMLSerializer().serializeToString(document);
```

Comment:
This will re-press the doc (with DOMParser), so maybe make a bit more changes to directly use the document?